File: Examples/CSharp/Knowledge-Base/CreatingDiagramWithVSTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Visio = Microsoft.Office.Interop.Visio;

namespace CSharp.Knowledge_Base
{
    public class CreatingDiagramWithVSTO
    {
        public static void Run() 
        {
            //ExStart:CreatingDiagramWithVSTO
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_KnowledgeBase();

            Visio.Application vdxApp = null;
            Visio.Document vdxDoc = null;
            try
            {
                //Create Visio Application Object
                vdxApp = new Visio.Application();

                //Make Visio Application Invisible
                vdxApp.Visible = false;

                //Create a new diagram
                vdxDoc = vdxApp.Documents.Add("");

                //Load Visio Stencil
                Visio.Documents visioDocs = vdxApp.Documents;
                Visio.Document visioStencil = visioDocs.OpenEx("Basic Shapes.vss",
                    (short)Microsoft.Office.Interop.Visio.VisOpenSaveArgs.visOpenHidden);

                //Set active page
                Visio.Page visioPage = vdxApp.ActivePage;

                //Add a new rectangle shape
                Visio.Master visioRectMaster = visioStencil.Masters.get_ItemU(@"Rectangle");
                Visio.Shape visioRectShape = visioPage.Drop(visioRectMaster, 4.25, 5.5);
                visioRectShape.Text = @"Rectangle text.";

                //Add a new star shape
                Visio.Master visioStarMaster = visioStencil.Masters.get_ItemU(@"Star 7");
                Visio.Shape visioStarShape = visioPage.Drop(visioStarMaster, 2.0, 5.5);
                visioStarShape.Text = @"Star text.";

                //Add a new hexagon shape
                Visio.Master visioHexagonMaster = visioStencil.Masters.get_ItemU(@"Hexagon");
                Visio.Shape visioHexagonShape = visioPage.Drop(visioHexagonMaster, 7.0, 5.5);
                visioHexagonShape.Text = @"Hexagon text.";


                //Save diagram as VDX
                vdxDoc.SaveAs(dataDir + "CreatingDiagramWithVSTO_Out.vdx");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                //Close active document and quit
                vdxDoc.Close();
                vdxApp.Quit();
            }
            //ExEnd:CreatingDiagramWithVSTO
        }
    }
}

File: Examples/VisualBasic/Knowledge-Base/CreatingDiagramWithVSTO.vb

Imports Aspose.Diagram
Imports System
Imports System.IO
Imports Visio = Microsoft.Office.Interop.Visio
Imports System.Collections

Public Class CreatingDiagramWithVSTO
    Public Shared Sub Run()
        ' ExStart:CreatingDiagramWithVSTO
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_KnowledgeBase()

        Dim vdxApp As Visio.Application = Nothing
        Dim vdxDoc As Visio.Document = Nothing
        Try
            ' Create Visio Application Object
            vdxApp = New Visio.Application()

            ' Make Visio Application Invisible
            vdxApp.Visible = False

            ' Create a new diagram
            vdxDoc = vdxApp.Documents.Add("")

            ' Load Visio Stencil
            Dim visioDocs As Visio.Documents = vdxApp.Documents
            Dim visioStencil As Visio.Document = visioDocs.OpenEx("Basic Shapes.vss", CShort(Microsoft.Office.Interop.Visio.VisOpenSaveArgs.visOpenHidden))

            ' Set active page
            Dim visioPage As Visio.Page = vdxApp.ActivePage

            ' Add a new rectangle shape
            ' Dim visioRectMaster As Visio.Master = visioStencil.Masters.get_ItemU("Rectangle")
            ' Dim visioRectShape As Visio.Shape = visioPage.Drop(visioRectMaster, 4.25, 5.5)
            ' visioRectShape.Text = "Rectangle text."

            ' Add a new star shape
            ' Dim visioStarMaster As Visio.Master = visioStencil.Masters.get_ItemU("Star 7")
            ' Dim visioStarShape As Visio.Shape = visioPage.Drop(visioStarMaster, 2.0, 5.5)
            ' visioStarShape.Text = "Star text."

            ' Add a new hexagon shape
            ' Dim visioHexagonMaster As Visio.Master = visioStencil.Masters.get_ItemU("Hexagon")
            ' Dim visioHexagonShape As Visio.Shape = visioPage.Drop(visioHexagonMaster, 7.0, 5.5)
            ' visioHexagonShape.Text = "Hexagon text."


            ' Save diagram as VDX
            vdxDoc.SaveAs(dataDir & Convert.ToString("CreatingDiagramWithVSTO_Out.vdx"))

        Catch ex As Exception
            Console.WriteLine(ex.Message)
            Console.WriteLine("This example will only work if you apply a valid Aspose License. You can purchase full license or get 30 day temporary license from http://www.aspose.com/purchase/default.aspx.")
        End Try

    End Sub
End Class
